code/script.js

```diff
@@ -26,14 +37,195 @@ const showMessage = (message, sender) => {
+}
+
+const handleNameInput = document.getElementById('name-form').addEventListener('submit', (event) => {
+  event.preventDefault()
+  const nameInput = document.getElementById('name-input')
+  const userName = nameInput.value
+
+  userReply(userName, 'user')
+  nameInput.value = ''
+  setTimeout(() => bookOrCancel(userName), 1000)
+}
+)
+
+// Question2 (booking or canceling)
+
+const bookOrCancel = (msg) => {
+  questionNumber++
+  botReply(`Hello ${msg}! Do you wish to book or cancel an appointment?`)
+
+  inputWrapper.innerHTML = `
+  <button id="bookBtn">Book</button>
+  <button id="cancelBtn">Cancel</button>
+`
+
+  document
+    .getElementById('bookBtn')
+    .addEventListener('click', () => nextQuestion(`I want to book an appointment.`))
+  document
+    .getElementById('cancelBtn')
+    .addEventListener('click', () => nextQuestion(`I want to cancel an appointment.`))
+}
+
+// Question3 (choose treatment)
+
+const showMenu = (type) => {
+  questionNumber++
+
+  if (type === `I want to book an appointment.`) {
+    botReply(`Awesome! What kind of appointment would you like to book?`)
+    inputWrapper.innerHTML = `
+        <select id="select">
+          <option value="" selected disabled>Select appointment</option>
+          <option value="Haircut">Haircut</option>
+          <option value="Coloring">Coloring</option>
+          <option value="Hair treatment">Hair Treatment</option>
+          <option value="Perm">Perm</option>
+        </select>
+      `
+  } else {
+    botReply(`Please call 08-123 45 67 to cancel your appointment. Sorry to see you go!`)
+    inputWrapper.innerHTML = ``
+  }
+
+  const select = document.getElementById('select')
+  select.addEventListener('change', () => nextQuestion(select.value))
+}
+
+// Question4 (choose hairdresser)
+
+const showHairdresser = (treatment) => {
+  questionNumber++
+
+  botReply(`A ${treatment.toLowerCase()}, how exciting! Which hairdresser would you like to book?`)
+
+  inputWrapper.innerHTML = `
+  <button id="bookAnna">Anna</button>
+  <button id="bookHaru">Haru</button>
+`
+  document
+    .getElementById('bookAnna')
+    .addEventListener('click', () => nextQuestion(`Anna`))
+  document
+    .getElementById('bookHaru')
+    .addEventListener('click', () => nextQuestion(`Haru`))
+}
+
+// Question5 (choose date)
+
+const showDate = (hairdresser) => {
+  questionNumber++
+  botReply(`${hairdresser} is available for booking! Which date do you want to come?`)
+
+  //Variables to prevent choosing past dates.
+  const currentDate = new Date()
+  const formattedDate = currentDate.toISOString().split('T')[0]
+
+  inputWrapper.innerHTML = `
+    <form id="input">
+      <label for="date"></label>
+      <input type="date" id="date" name="date" min=${formattedDate}>
+    </form>
+    `
+  const input = document.getElementById('date')
+  input.addEventListener('change', () => {
+    appointmentDate = input.value;
+    nextQuestion(input.value);
+  })
```

Comment:
Nice input addition. Adding a date and not only dropdown/buttons. 
Also very well done making the last dates not available.